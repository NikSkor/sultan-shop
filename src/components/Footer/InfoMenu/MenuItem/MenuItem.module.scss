.itemBtn {
  font-weight: 400;
  font-size: 14px;
  line-height: 150%;
  transition: all .3s ease-in-out;

  &:hover {
    color: #E5E5E5;
  }
}