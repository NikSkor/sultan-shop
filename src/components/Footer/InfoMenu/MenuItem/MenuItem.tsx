import React, { FC } from 'react';
import style from './MenuItem.module.scss';

interface MenuOneItem {
  item: string,
}

const MenuItem: FC<MenuOneItem> = ({item}) => {
  return (
    <li className={style.item}>
        <button className={style.itemBtn}>{item}</button>
      </li>
  )
}

export default MenuItem;