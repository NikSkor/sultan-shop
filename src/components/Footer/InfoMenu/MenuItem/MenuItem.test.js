import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MenuItem from './MenuItem.tsx';

describe('MenuItem.tsx', () => {
  afterEach(cleanup);
  test('component renders props correctly, truthy values', () => {
    const tree = renderer
      .create(<MenuItem key='1' item='name' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderer.create(<MenuItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('there is the button', () => {
    render(<MenuItem />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });
});
