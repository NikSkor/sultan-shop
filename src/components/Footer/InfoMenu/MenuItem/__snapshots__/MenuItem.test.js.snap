// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem.tsx component renders props correctly, falsy values 1`] = `
<li
  className="item"
>
  <button
    className="itemBtn"
  />
</li>
`;

exports[`MenuItem.tsx component renders props correctly, truthy values 1`] = `
<li
  className="item"
>
  <button
    className="itemBtn"
  >
    name
  </button>
</li>
`;

exports[`MenuItem.tsx there is the button 1`] = `
<button
  class="itemBtn"
/>
`;
